pkg/reconciler/status.go

```diff
@@ -75,26 +84,21 @@ func (s *Status) SetReconciling(meta metav1.ObjectMeta) {
 
 	s.S.FriendlyDescription = "Reconciling"
 	s.S.UsefulErrorMessage = ""
-
-	s.UpdateFunc(s.S)
 }
 
 func (s *Status) SetDeleting(meta metav1.ObjectMeta) {
 	s.markObservedLatest(meta)
-	s.removeAllConditions()
 
 	s.S.Conditions = append(s.S.Conditions, kcv1alpha1.Condition{
 		Type:   kcv1alpha1.Deleting,
 		Status: corev1.ConditionTrue,
 	})
 
 	s.S.FriendlyDescription = "Deleting"
-
-	s.UpdateFunc(s.S)
 }
 
 func (s *Status) SetReconcileCompleted(err error) {
```

Comment:
it seems that we are modifying generic pkg/reconciler. im guessing changes would affect other CRs like App CR. it seems that just removal of reconciling vs everything would cause problems, right?